src/pages/ResultsPage.js

```diff
@@ -188,6 +146,82 @@ class BaseResultsPage extends BaseCategoriesPage {
             }
         }
     }
+
+    onBackClick(event: SyntheticInputEvent): void {
+        if (this.refs.component.onGoBack) {
+            this.refs.component.onGoBack(event)
+        }
+        if (!event.defaultPrevented) {
+            this.props.history.goBack();
+        }
+    }
+
+    render(): ReactElement {
+        const Component = this.component();
+
+        return (
+            <div className="ResultsPage">
+                <AppBar
+                    title={this.title}
+                    onBackTouchTap={this.onBackClick.bind(this)}
+                    onForwardTouchTap={() =>
+                        this.props.history.pushState(
+                            null,
+                            "/",
+                            {}
+                        )
+                    }
+                    forwardIcon={<icons.House className="small" />}
+                />
+
+                <Component
+                    ref="component"
+                    {...this.state}
+                    {...this.props}
+                    category={this.category}
+                    search={this.props.params.search}
+                    loadMore={this.renderLoadMore()}
+                    title={this.title}
+                    loading={this.loading}
+                />
+            </div>
+        );
+    }
+
+    renderLoadMore(): ?ReactElement {
+        if (this.state.meta && this.state.meta.next) {
+            return (
+                <ButtonListItem
+                    className="MoreResultsButton"
+                    primaryText="Load more results…"
+                    onTouchTap={this.loadMore.bind(this)}
+                />
+            );
+        }
+
+        if (this.loading) {
+            return (
+                <div className="progress">
+                    <icons.Loading />
+                </div>
+            );
+        }
+    }
+
 }
 
-export default BaseResultsPage;
+export default ResultsPage;
+
+export class ResultsPageListing extends ResultsPage {
+    // flow:disable until flow can track inheritance for ReactComponent
+    component(): ReactComponent {
```

Comment:
I went with instance in this case since it made the caller read better, but static would also make sense.